src/Handler/FpmHandler.php

```diff
@@ -172,59 +172,23 @@ private function isReady(): bool
         return file_exists(self::SOCKET);
     }
 
-    private function eventToFastCgiRequest(array $event): ProvidesRequestData
+    private function eventToFastCgiRequest(HttpRequestEvent $event): ProvidesRequestData
     {
-        $requestBody = $event['body'] ?? '';
-        if ($event['isBase64Encoded'] ?? false) {
-            $requestBody = base64_decode($requestBody);
-        }
-
-        $method = strtoupper($event['httpMethod']);
-        $request = new FastCgiRequest($method, $this->handler, $requestBody);
-
-        $queryString = $this->getQueryString($event);
-        $uri = $event['path'] ?? '/';
-        if (! empty($queryString)) {
-            $uri .= '?' . $queryString;
-        }
-
-        $protocol = $event['requestContext']['protocol'] ?? 'HTTP/1.1';
-
-        // Normalize headers
-        if (isset($event['multiValueHeaders'])) {
-            $headers = $event['multiValueHeaders'];
-        } else {
-            $headers = $event['headers'] ?? [];
-            // Turn the headers array into a multi-value array to simplify the code below
-            $headers = array_map(function ($value): array {
-                return [$value];
-            }, $headers);
-        }
-        $headers = array_change_key_case($headers, CASE_LOWER);
```

Comment:
All of that logic (that was specific to FPM before) has been moved into the `HttpRequestEvent` class.

This allows to avoid the code duplication that I mentioned in #446 (to support PSR-7 and Symfony requests).